app/models/instance.rb

```diff
@@ -23,4 +64,10 @@ def to_param
   def cache_key
     domain
   end
+
+  private
+
+  def cache(key)
+    Rails.cache.fetch("instances/#{cache_key}/#{key}") { yield }
```

Comment:
I'd still use an explicit short-lived timeout for that. And/or explicitly invalidate it on certain actions.

Reply:
Alright I removed the caching of those values, let's see how these improvements fare on their own